block-layer/codecs/dag-cbor.md

```diff
@@ -37,93 +39,103 @@ In DAG-CBOR, map keys must be strings, as defined by the [IPLD Data Model]. Othe
+6. Encode and decode must operate on a single top-level CBOR object. Back-to-back concatenated objects are not allowed or supported, as suggested by [section 5.1 of RFC 8949] for _streaming applications_. All bytes of an encoded DAG-CBOR object must decode to a single object. Extraneous bytes included in an IPLD block, whether valid or invalid CBOR, must not be accepted as valid DAG-CBOR.
 
 ## Implementations
 
 ### JavaScript
 
-[dag-cbor], used by [ipld] and [@ipld/block] adheres to this specification, with the following caveats:
+**[@ipld/dag-cbor]**, for use with [multiformats] adheres to this specification, with the following caveats:
+ * Complete strictness is not yet enforced on decode. Specifically: correct map ordering is not enforced and floats that are not encoded as 64-bit are not rejected.
+ * [`BigInt`] is accepted along with `Number` for encode, but the smallest-possible rule is followed when encoding. When decoding integers outside of the JavaScript "safe integer" range, a [`BigInt`] will be used.
+
+The legacy **[ipld-dag-cbor]** implementation adheres to this specification, with the following caveats:
 
- * Strictness is not yet enforced on decode, blocks encoded that don't follow the strictness rules are not rejected
- * Floating point values are encoded as their smallest form rather than always double-precision.
- * Many additional object types outside of the Data Model are currently accepted for encoding.
- * IEEE 754 special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
+ * Strictness is not enforced on decode; blocks encoded that do not follow the strictness rules are not rejected.
+ * Floating point values are encoded as their smallest form rather than always 64-bit.
+ * Many additional object types outside of the Data Model are currently accepted for decode and encode, including `undefined`.
+ * [IEEE 754] special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
+ * Integers outside of the JavaScript "safe integer" range will use the third-party [bignumber.js] library to represent their values.
+
+Note that inability to clearly differentiate between integers and floats in JavaScript may cause problems with round-trips of floating point values. See the [IPLD Data Model] and the discussion on [Limitations](#limitations) below for further discussion on JavaScript numbers and recommendations regarding the use of floats.
 
 ### Go
 
-[ipld-cbor] and [ipld-prime] adhere to this specification, with the following caveats:
+**[go-ipld-cbor]** and **[go-ipld-prime]** adhere to this specification, with the following caveats:
 
- * Strictness is not yet enforced on decode, blocks encoded that don't follow the strictness rules are not rejected
- * IEEE 754 special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
+ * Strictness is not enforced on decode; blocks encoded that do not follow the strictness rules are not rejected.
```

Comment:
out of curiosity, do we have an issue about this for ipld-prime?

Reply:
Probably need to break this down into a couple things:

- The NaN/Infinite issue
- Float encoding size (I think we accept variable size, but only emit 64bit size, iirc)
- Map sorting (but [I have opinions about whether this is a good idea](https://github.com/ipld/specs/pull/353/files#r561683154))